### ISSUE TITLE
Pass additional data fields to <option> tags in <select>s

### ISSUE THREAD

**Issue by fxgallego** [TrestleAdmin/trestle#204]
There is a way (in classic rails forms) to pass extra data in options of a select tag.
Each option can be written as:
`[option_value, option_label, {}, {my_extra_data: 'my_extra_data_value'}]`

The problem is Trestle `Form::Fields::Select::Choices` is filtering extra hashes in the option array in this method:
```ruby
       def option_text_and_value(option)
            if !option.is_a?(String) && option.respond_to?(:first) && option.respond_to?(:last)
              option = option.reject { |e| Hash === e } if Array === option
              [option.first, option.last]
              option
            elsif option.respond_to?(:id)
              [Trestle::Display.new(option).to_s, option.id]
            else
              [option, option]
            end
          end
```
The second line is the problematic one. I can't realize why is this line there but I commented the code of that line and the next one and now I can pass the additional options data and I'm not experiencing any problems so far.

**Comment by spohlenz** [comment]
For now, you can wrap your options with a call to `options_for_select` to bypass the additional handling of options.

```ruby
select :my_attr, options_for_select([["Label", 1, { data: { foo: "bar" }]])
```

The intention for this code is to allow a scope or array of objects to be passed that will be automatically formatted for the select options. However it is clear that it does need some additional work (and some tests).

**Comment by spohlenz** [comment]
This should now be fixed in da2860c9131907acd781f484451d430d4f791966, along with some much better specs for select and other form fields.
